### ISSUE TITLE
Tooltip component does not work with disabled Button component

### ISSUE THREAD

**Issue by ecgan** [WordPress/gutenberg#27451]
**Describe the bug**

[Tooltip](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/tooltip) component does not work with disabled [Button](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/button) component:

**Steps to reproduce:**

```jsx
/**
 * External dependencies
 */
import { Tooltip, Button } from '@wordpress/components';

const MyComponent = () => {
	return (
		<Tooltip text="Show helpful text here">
			<Button disabled>Click me</Button>
		</Tooltip>
	);
};

export default MyComponent;
```

1. Create a React component like the above MyComponent.
1. Use the component in your React page. 
1. Open up a browser, navigate to the React page, hover your mouse over the disabled "Click me" button. **Expect the tooltip to show up, but it does not.**
1. Remove the `disabled` prop, and reload the page. The tooltip will show up as expected.

**Expected behavior**

Tooltip should appear when I hover over the disabled button.

**Comment by grzim** [comment]
@ecgan, I think I have found the solution to this problem. Can you please write what are the steps you did to see this issue, so I can test it in the same scenario?

**Comment by ecgan** [comment]
@grzim , I have updated the description with more details. Let me know if you need more info.

**Comment by draganescu** [comment]
This does not appear to be a bug, as @grzim noted, this was expressly tested against. I don't have enough context to say much, but I added some labels to get attention here.
